cdc/entry/schema_storage.go

```diff
@@ -59,6 +56,48 @@ func (s *SingleSchemaSnapshot) HandleDDL(job *timodel.Job) error {
 	return s.handleDDL(job)
 }
 
+// PreTableInfo returns the table info which with overwrite by the specified job
+func (s *SingleSchemaSnapshot) PreTableInfo(job *timodel.Job) (*model.TableInfo, error) {
+	switch job.Type {
+	case timodel.ActionCreateSchema, timodel.ActionModifySchemaCharsetAndCollate, timodel.ActionDropSchema:
+		return nil, nil
+	case timodel.ActionCreateTable, timodel.ActionCreateView, timodel.ActionRecoverTable:
+		// no pre table info
+		return nil, nil
+	case timodel.ActionRenameTable, timodel.ActionDropTable, timodel.ActionDropView, timodel.ActionTruncateTable:
+		// get the table will be dropped
+		table, ok := s.TableByID(job.TableID)
+		if !ok {
+			return nil, errors.NotFoundf("table %d", job.TableID)
+		}
+		return table, nil
+	case timodel.ActionTruncateTablePartition, timodel.ActionAddTablePartition, timodel.ActionDropTablePartition:
+		tableID := job.BinlogInfo.TableInfo.ID
+		table, ok := s.TableByID(tableID)
```

Comment:
Is it sure to return the table that the partition belongs to, if so, seems the logic is similar to the default branch